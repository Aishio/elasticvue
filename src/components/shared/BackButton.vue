<template>
  <v-btn :to="route" class="ml-a">
    <v-icon>{{icon}}</v-icon>
    {{text}}
  </v-btn>
</template>

<script>
  export default {
    name: 'BackButton',
    props: {
      route: {
        type: Object,
        default: () => {
          return {
            name: 'Home',
            params: {}
          }
        }
      },
      text: {
        type: String,
        default: 'Back'
      },
      icon: {
        type: String,
        default: 'mdi-chevron-left'
      }
    }
  }
</script>
