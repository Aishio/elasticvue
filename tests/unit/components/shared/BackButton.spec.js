import Vuetify from 'vuetify'
import VBtn from 'vuetify/es5/components/VBtn'
import VIcon from 'vuetify/es5/components/VIcon'
import BackButton from '@/components/shared/BackButton'
import { createLocalVue, mount, RouterLinkStub } from '@vue/test-utils'

describe('components/shared/BackButton.vue', () => {
  it('should render correct default contents', () => {
    let localVue = createLocalVue()
    localVue.use(Vuetify, { components: { VBtn, VIcon } })

    const wrapper = mount(BackButton, {
      localVue: localVue,
      stubs: {
        'router-link': RouterLinkStub
      }
    })

    expect(wrapper.element).toMatchSnapshot()
  })
})
