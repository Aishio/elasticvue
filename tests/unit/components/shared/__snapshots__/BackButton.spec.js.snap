// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/shared/BackButton.vue should render correct default contents 1`] = `
<a
  class="ml-a v-btn v-btn--contained v-btn--router theme--light v-size--default"
>
  <span
    class="v-btn__content"
  >
    <i
      aria-hidden="true"
      class="v-icon notranslate mdi mdi-chevron-left theme--light"
    />
    
  Back

  </span>
</a>
`;
